lib/assert.js

```diff
@@ -61,11 +61,7 @@ assert.AssertionError = function AssertionError(options) {
 util.inherits(assert.AssertionError, Error);
 
 function truncate(s, n) {
-  if (typeof s === 'string') {
-    return s.length < n ? s : s.slice(0, n);
-  } else {
-    return s;
-  }
+  return s.length < n ? s : s.slice(0, n);
```

Comment:
Not sure. My guess is it was done this way as a micro-optimization for the common case. Seems like just calling `s.slice(0,n)` as you suggest would be better.